File: topics/control-flow/exercises/mouse_pointer.c
#define LARGE_RADIUS 6
#define SMALL_RADIUS 3

void draw_cursor()
{
    float dot_x, dot_y;
    hide_mouse();
    
    while( ! window_close_requested() )
    {
        process_events();
        
        dot_x = mouse_x();
        dot_y = mouse_y();
        
        clear_screen(ColorWhite);
        draw_framerate(0,0);
        draw_circle(ColorBlack, dot_x, dot_y, LARGE_RADIUS);
        
        if ( mouse_down(LEFT_BUTTON) )
            fill_circle(ColorBlack, dot_x, dot_y, SMALL_RADIUS);
        else
            draw_circle(ColorBlack, dot_x, dot_y, SMALL_RADIUS);
        
        refresh_screen();
    }
    
    show_mouse();
}

File: topics/control-flow/exercises/MousePointer.pas
const LARGE_RADIUS = 6;
const SMALL_RADIUS = 3;

procedure DrawCursor();
var
    dotX, dotY: Single;
begin
    HideMouse();
    
    while not WindowCloseRequested() do
    begin
        ProcessEvents();
        
        dotX = MouseX();
        dotY = MouseY();
        
        ClearScreen(ColorWhite);
        DrawFramerate(0,0);
        DrawCircle(ColorBlack, dotX, dotY, LARGE_RADIUS);
        
        if ( MouseDown(LEFT_BUTTON) )
            fill_circle(ColorBlack, dotX, dotY, SMALL_RADIUS)
        else
            DrawCircle(ColorBlack, dotX, dotY, SMALL_RADIUS);
        
        RefreshScreen();
    end;
    
    ShowMouse();
end;